src/unix/kqueue.c

```diff
@@ -302,9 +333,19 @@ void uv__io_poll(uv_loop_t* loop, int timeout) {
     have_signals = 0;
     nevents = 0;
 
+#if UV__KQUEUE_EVFILT_USER
+    /* EVFILT_USER event requires no extra file descriptor,
+     * thus we should allow empty watcher list during polling.
+     */
```

Comment:
Instead of needing to make these conditional, I'd recommend sticking a `maybe_resize(loop, 30)` call during loop initialization. It is a bit silly to need to grow this several times when adding the first couple fd anyways